rpcserver.go

```diff
@@ -2073,6 +2081,82 @@ func (r *rpcServer) OpenChannelSync(ctx context.Context,
 	}
 }
 
+// BatchOpenChannel attempts to open multiple single-funded channels in a
+// single transaction in an atomic way. This means either all channel open
+// requests succeed at once or all attempts are aborted if any of them fail.
+// This is the safer variant of using PSBTs to manually fund a batch of
+// channels through the OpenChannel RPC.
+func (r *rpcServer) BatchOpenChannel(ctx context.Context,
+	in *lnrpc.BatchOpenChannelRequest) (*lnrpc.BatchOpenChannelResponse,
+	error) {
+
+	if err := r.canOpenChannel(); err != nil {
+		return nil, err
+	}
+
+	// We need the wallet kit server to do the heavy lifting on the PSBT
+	// part. Doesn't make sense to re-implement all of that in the main
+	// server again.
+	var walletKitServer walletrpc.WalletKitServer
+	for _, subServer := range r.subServers {
+		if subServer.Name() == walletrpc.SubServerName {
+			walletKitServer = subServer.(walletrpc.WalletKitServer)
```

Comment:
A subserver is an RPC server that can be enabled/disabled using build tags. They were originally meant as a way for us to introduce new, more experimental RPCs and hide them behind build flags before they were declared as stable. But since we now enable all subservers by default in our release builds the "experimental" state can't really be claimed anymore. So it's more of a way to split the RPC code into smaller chunks now.